src/quota_operation_aggregator_test.cc

```diff
@@ -0,0 +1,187 @@
+  QuotaOperationAggregator aggregator(operation1_);
+
+  QuotaOperation operation = aggregator.ToOperationProto();
+  ASSERT_TRUE(aggregator.is_aggregated());
+
+  std::set<std::pair<std::string, int>> quota_metrics;
+  for (auto quota_metric : operation.quota_metrics()) {
+    quota_metrics.insert(
+        std::make_pair(quota_metric.metric_name(),
+                       quota_metric.metric_values(0).int64_value()));
+  }
+
+  std::set<std::pair<std::string, int>> expected_costs = {{"metric_first", 1},
+                                                          {"metric_second", 1}};
+  ASSERT_EQ(quota_metrics, expected_costs);
+}
+
+TEST_F(QuotaOperationAggregatorImplTest, TestInitializationNotAggregated) {
+  QuotaOperationAggregator aggregator(operation3_);
+
+  QuotaOperation operation = aggregator.ToOperationProto();
+  ASSERT_FALSE(aggregator.is_aggregated());
+}
+
+TEST_F(QuotaOperationAggregatorImplTest, TestMergeOperation) {
+  QuotaOperationAggregator aggregator(operation1_);
+
+  QuotaOperation operation = aggregator.ToOperationProto();
+  ASSERT_TRUE(aggregator.is_aggregated());
+
+  std::set<std::pair<std::string, int>> quota_metrics;
+  for (auto quota_metric : operation.quota_metrics()) {
+    quota_metrics.insert(
+        std::make_pair(quota_metric.metric_name(),
+                       quota_metric.metric_values(0).int64_value()));
+  }
+
+  std::set<std::pair<std::string, int>> expected_costs = {{"metric_first", 1},
+                                                          {"metric_second", 1}};
+  ASSERT_EQ(quota_metrics, expected_costs);
+
+  aggregator.MergeOperation(operation2_);
+
+  operation = aggregator.ToOperationProto();
+  ASSERT_TRUE(aggregator.is_aggregated());
+
+  quota_metrics.clear();
+  for (auto quota_metric : operation.quota_metrics()) {
+    quota_metrics.insert(
+        std::make_pair(quota_metric.metric_name(),
+                       quota_metric.metric_values(0).int64_value()));
+  }
+
+  expected_costs = {{"metric_first", 3}, {"metric_second", 4}};
+  ASSERT_EQ(quota_metrics, expected_costs);
+}
+
+TEST_F(QuotaOperationAggregatorImplTest, TestMergeOperationToNullOperation) {
+  QuotaOperationAggregator aggregator(operation3_);
+  ASSERT_FALSE(aggregator.is_aggregated());
+
+  aggregator.MergeOperation(operation2_);
+  ASSERT_TRUE(aggregator.is_aggregated());
+
+  QuotaOperation operation = aggregator.ToOperationProto();
+
+  std::set<std::pair<std::string, int>> quota_metrics;
+  for (auto quota_metric : operation.quota_metrics()) {
+    quota_metrics.insert(
+        std::make_pair(quota_metric.metric_name(),
+                       quota_metric.metric_values(0).int64_value()));
+  }
+
+  std::set<std::pair<std::string, int>> expected_costs = {{"metric_first", 2},
+                                                          {"metric_second", 3}};
+  ASSERT_EQ(quota_metrics, expected_costs);
+}
+
+TEST_F(QuotaOperationAggregatorImplTest, TestMergeOperationNullOperation) {
```

Comment:
I wanted to test message without quota_metrics. 
It is not possible in real case. I removed it.